concordium-consensus/src/Concordium/GlobalState/Persistent/Account/StructureV1.hs

```diff
@@ -0,0 +1,1314 @@
+migratePersistentAccountStakeEnduring ::
+    (SupportMigration m t, IsAccountVersion av) =>
+    PersistentAccountStakeEnduring av ->
+    t m (PersistentAccountStakeEnduring av)
+migratePersistentAccountStakeEnduring PersistentAccountStakeEnduringNone =
+    return PersistentAccountStakeEnduringNone
+migratePersistentAccountStakeEnduring PersistentAccountStakeEnduringBaker{..} = do
+    newBakerInfo <- migrateReference return paseBakerInfo
+    return $!
+        PersistentAccountStakeEnduringBaker
+            { paseBakerInfo = newBakerInfo,
+              ..
+            }
+migratePersistentAccountStakeEnduring PersistentAccountStakeEnduringDelegator{..} =
+    return $! PersistentAccountStakeEnduringDelegator{..}
+
+
+-- |This relies on the fact that the 'AccountV2' hashing of 'AccountStake' is independent of the
+-- staked amount.
+instance (MonadBlobStore m) => MHashableTo m (AccountStakeHash 'AccountV2) (PersistentAccountStakeEnduring 'AccountV2) where
+    getHashM stake = getHash <$> persistentToAccountStake stake 0
+
+-- * Enduring account data
+
+-- |Enduring data associated with an account. This is data that does not change very often.
+-- The 'AccountMerkleHash' is computed and stored for this data so that we can avoid deserializing
+-- elements of it unnecessarily.
+--
+-- The persisting account data is stored under an 'EagerBufferedRef' as this includes the account
+-- keys, which are required for any transaction originating from the account, and so are loaded
+-- eagerly.
+--
+-- The encrypted amount is 'Nullable' so that accounts with no encrypted balance can be represented
+-- succinctly.  The encrypted balance is stored as a 'LazyBufferedRef', which is not automatically
+-- cached with the account, since loading the encrypted balance is relatively expensive and likely
+-- not necessary for many operations.
+--
+-- The release schedule is likewise 'Nullable' and stored under a 'LazyBufferedRef' for similar reasons.
+-- However, as well as the 'LazyBufferedRef', we store the locked balance (if it is non-trivial).
+-- This is because we typically require ready access to the locked balance so that we can compute
+-- the available balance.
+--
+-- The stake is not stored under a reference (excepting the baker info, as per the definition of
+-- 'PersistentAccountStakeEnduring'). This is since the information is relatively succinct.
+data PersistentAccountEnduringData (av :: AccountVersion) = PersistentAccountEnduringData
+    { -- |The Merkle hash computed from the other fields.
+      paedHash :: !(AccountMerkleHash av),
+      -- |A reference to the persisting account data.
+      paedPersistingData :: !(EagerBufferedRef PersistingAccountData),
+      -- |The encrypted amount. Invariant: if this is present, it will not satisfy
+      -- 'isZeroAccountEncryptedAmount'.
+      paedEncryptedAmount :: !(Nullable (LazyBufferedRef PersistentAccountEncryptedAmount)),
```

Comment:
Check that this is properly guaranteed. An account can end in this state even after having used encrypted amounts.

Reply:
I have revised this to `isInitialPersistentAccountEncryptedAmount` and it is checked and preserved.